central/networkgraph/flow/datastore/internal/store/postgres/tests/flow_impl_benchmark_test.go

```diff
@@ -101,6 +107,34 @@ func BenchmarkGetExternalFlows(b *testing.B) {
 	b.Run("deployment with 1000 ingress and 1000 egress flows", benchmarkGetExternalFlows(flowStore, fixtureconsts.Deployment3))
 }
 
+func BenchmarkPruneOrphanedFlowsForDeployment(b *testing.B) {
+	psql := pgtest.ForT(b)
+
+	clusterStore := postgresFlowStore.NewClusterStore(psql)
+	flowStore, err := clusterStore.CreateFlowStore(context.Background(), fixtureconsts.Cluster1)
+	require.NoError(b, err)
+	eStore := entityStore.GetTestPostgresDataStore(b, psql)
+
+	b.Run("1000 flows to be pruned", benchmarkPruneOrphanedFlowsForDeployment(flowStore, eStore, fixtureconsts.Deployment1, 1000))
+	b.Run("10000 flows to be pruned", benchmarkPruneOrphanedFlowsForDeployment(flowStore, eStore, fixtureconsts.Deployment1, 10000))
+	b.Run("100000 flows to be pruned", benchmarkPruneOrphanedFlowsForDeployment(flowStore, eStore, fixtureconsts.Deployment1, 100000))
+	b.Run("900000 flows to be pruned", benchmarkPruneOrphanedFlowsForDeployment(flowStore, eStore, fixtureconsts.Deployment1, 900000))
```

Comment:
Will not merge the tests with 900K rows, but it is good for testing to check that the pruner can handle a large number of rows.

Reply:
What was the timeout when this failed?  I suspect the context has a single timeout that ALL calls must finish within.  Since we are stacking MANY calls, maybe we need to rethink that a little bit.

Reply:
The failure is shown in the PR description in the testing section. For both pruning by deployment and update time the mutex timed out after 10 seconds. This is resolved by locking and unlocking the mutex multiple times, enabling other threads to safely do operations on the tables during the pruning.